src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```
@@ -122,8 +144,40 @@ public static void distributedMismatch(LfNetwork network, double mismatch, LoadF
         }
     }
 
+    public static void getAllSwitchesToOperate(Network network, List<Action> actions, Set<Switch> allSwitchesToClose, Set<Switch> allSwitchesToOpen) {
+        actions.stream().filter(action -> action.getType().equals(SwitchAction.NAME))
+                .map(action -> ((SwitchAction) action).getSwitchId())
+                .forEach(id -> {
+                    Switch sw = network.getSwitch(id);
+                    if (sw.isOpen()) {
+                        allSwitchesToClose.add(sw);
+                    } else {
+                        allSwitchesToOpen.add(sw);
+                    }
+                });
+    }
+
+    private static Map<String, LfAction> createLfActions(LfNetwork network, List<Action> actions) {
+        return actions.stream().collect(Collectors.toMap(Action::getId, action -> new LfAction(action, network)));
+    }
+
+    private static Map<String, List<OperatorStrategy>> indexOperatorStrategiesByContingencyId(List<PropagatedContingency> propagatedContingencies,
+                                                                                              List<OperatorStrategy> operatorStrategies) {
+        Set<String> contingencyIds = propagatedContingencies.stream().map(propagatedContingency -> propagatedContingency.getContingency().getId()).collect(Collectors.toSet());
+        Map<String, List<OperatorStrategy>> operatorStrategiesByContingencyId = new HashMap<>();
+        for (OperatorStrategy operatorStrategy : operatorStrategies) {
+            if (contingencyIds.contains(operatorStrategy.getContingencyId())) {
+                operatorStrategiesByContingencyId.computeIfAbsent(operatorStrategy.getContingencyId(), key -> new ArrayList<>()).add(operatorStrategy);
+            } else {
+                LOGGER.warn("An operator strategy linked to Contingency {} that is not present in the list of Contingencies", operatorStrategy.getContingencyId());
```

I would throw an exception